src/main/java/jsr223/cpython/PythonScriptEngine.java

```diff
@@ -70,8 +71,11 @@
     public PythonScriptEngine() {
     }
 
-    private static synchronized GatewayServer startGatewayServer(EntryPoint entryPoint) {
-        GatewayServer gatewayServer = new GatewayServer(entryPoint, 0);
+    private static synchronized GatewayServer startGatewayServer(EntryPoint entryPoint, String authToken) {
+        GatewayServer gatewayServer = new GatewayServer.GatewayServerBuilder().entryPoint(entryPoint)
+                                                                              .authToken(authToken)
+                                                                              .javaPort(0)
```

Comment:
what is the purpose of ```javaPort(0)``` ?

Reply:
0 is a reserved keyword in py4j, which means that the next available port will be used.